### ISSUE TITLE
Fix 3.10 and 3.11 support

### ISSUE THREAD

**Issue by iccode17** [nbiotcloud/ucdp-amba#3]
```
    from tests.ahb_driver import AHBMasterDriver, AHBSlaveDriver, BurstType, SizeType
E     File "/home/runner/work/ucdp-amba/ucdp-amba/tests/ahb_driver.py", line 203
E       f"=MST WRITE= data: [{','.join(f"0x{x:0{(2**size) * 2}X}" for x in log_data)}] "
E                                         ^
E   SyntaxError: invalid hexadecimal literal
```

**Comment by markusdd** [comment]
do you have in mind what has changed here from .10/11 to 12?

this is standard hex formatting, not sure what broke here tbh

**Comment by markusdd** [comment]
Found the issue, they actually fixed inner quoting for f-strings only in 3.12.
https://peps.python.org/pep-0701/
